src/main/java/duke/data/WriteDataFile.java

```diff
@@ -0,0 +1,86 @@
+package duke.data;
+
+import duke.task.Task;
+import duke.task.Todo;
+import duke.task.Deadline;
+import duke.task.Event;
+
+import static duke.task.TaskHandler.tasks;
+
+import java.io.IOException;
+import java.io.FileWriter;
+
+public class WriteDataFile extends DataFile {
+    protected String textContent;
+
+    public WriteDataFile() {
+        executeFunction();
+    }
+
+    @Override
+    public void executeFunction() {
+        try {
+            setTextContent();
+            writeToFile();
+        } catch (IOException e) {
+            System.out.println("\tSomething went wrong: " + e.getMessage());
+        }
+    }
+
+    public void setTextContent() {
+        StringBuilder textBuild = new StringBuilder();
+        for(Task task : tasks) {
+            convertTaskToText(textBuild, task);
+        }
+        this.textContent = textBuild.toString();
+    }
+
+    public void convertTaskToText(StringBuilder textBuild, Task task) {
+        extractClass(textBuild, task);
+        textBuild.append(" | ");
+
+        extractIsDone(textBuild, task);
+        textBuild.append(" | ");
+        textBuild.append(task.getDescription());
+
+        if (task.getClass() != Todo.class) {
+            extractTime(textBuild, task);
+        }
+        textBuild.append(System.lineSeparator());
+    }
+
+    public void extractClass(StringBuilder textBuild, Task task) {
+        if(task.getClass() == Event.class) {
+            textBuild.append("E");
+        } else if(task.getClass() == Deadline.class) {
+            textBuild.append("D");
+        } else {
```

Comment:
perhaps it'll be good to check if it belongs to Todo class, rather than assume it'll fall into T class